src/test/optimizer/strategy/expression_reduction_rule_test.cpp

```diff
@@ -159,6 +161,127 @@ TEST_F(ExpressionReductionRuleTest, RewriteLikePrefixWildcard) {
   EXPECT_EQ(*expression_h, *greater_than_(s, "RED%"));
 }
 
+TEST_F(ExpressionReductionRuleTest, RemoveDuplicateAggregate) {
+  const auto table_definition = TableColumnDefinitions{{"a", DataType::Int, false}, {"b", DataType::Int, true}};
+  const auto table = Table::create_dummy_table(table_definition);
+  Hyrise::get().storage_manager.add_table("agg_table", table);
+  const auto stored_table_node = StoredTableNode::make("agg_table");
+
+  const auto col_a = lqp_column_({stored_table_node, ColumnID{0}});
+  const auto col_b = lqp_column_({stored_table_node, ColumnID{1}});
+
+  {
+    // SELECT SUM(a), COUNT(a), AVG(a) -> SUM(a), COUNT(a), SUM(a) / COUNT(a) AS AVG(a)
+    // clang-format off
+    const auto input_lqp = AggregateNode::make(expression_vector(), expression_vector(sum_(col_a), count_(col_a), avg_(col_a)),                                            // NOLINT
+                             stored_table_node);
+
+    const auto expected_aliases = std::vector<std::string>{"SUM(a)", "COUNT(a)", "AVG(a)"};
+    const auto expected_lqp = AliasNode::make(expression_vector(sum_(col_a), count_(col_a), div_(cast_(sum_(col_a), DataType::Double), count_(col_a))), expected_aliases,  // NOLINT
```

Comment:
It is not necessary, but we could add the same information regarding the cast as above?

Reply:
What do you mean?

Reply:
`// The cast will become unnecessary once #1799 is fixed`